receiver/signalfxreceiver/receiver.go

```diff
@@ -255,21 +253,26 @@ func (r *sfxReceiver) handleDatapointReq(resp http.ResponseWriter, req *http.Req
 		return
 	}
 
-	md, _ := signalFxV2ToMetricsData(r.logger, msg.Datapoints)
+	md, _ := signalFxV2ToMetrics(r.logger, msg.Datapoints)
 
 	if r.config.AccessTokenPassthrough {
 		if accessToken := req.Header.Get(splunk.SFxAccessTokenHeader); accessToken != "" {
-			if md.Resource == nil {
-				md.Resource = &resourcepb.Resource{}
+			for i := 0; i < md.ResourceMetrics().Len(); i++ {
+				rm := md.ResourceMetrics().At(i)
+				if rm.IsNil() {
+					rm.InitEmpty()
+				}
+
+				res := rm.Resource()
+				if res.IsNil() {
+					res.InitEmpty()
+				}
```

Comment:
if you rebase you will see that Resource cannot be null anymore (isNil is removed). and don't need to call InitEmpty as well.